packages/contracts-bedrock/scripts/upgrades/stage-1.4/README.md

```diff
@@ -0,0 +1,38 @@
+# Stage 1.4 Upgrade
+
+This directory contains a repeatable task for deploying and registering the stage 1.4 `FaultDisputeGame` with a
+`DisputeGameFactory` contract.
+
+## Dependencies
+
+- [`foundry`](https://getfoundry.sh/)
+- [`golang`](https://go.dev/doc/install)
+- [`jq`](https://jqlang.github.io/jq/download/)
+
+## Usage
+
+```sh
+# 1. Clone the monorepo, checkout the `op-contracts/v1.9.0-rc.3` tag, and navigate to this directory.
+git clone --branch op-contracts/v1.9.0-rc.3 git@github.com:ethereum-optimism/optimism.git && \
+  cd optimism/packages/contracts-bedrock/scripts/upgrades/stage-1.4
+
+# 2. Set up the `.env` file
+cp .env.example .env && vim .env
+
+# 3. Run the deployment task.
+#
+# This task will deploy a new `FaultDisputeGame` + `RISCV` VM + `DelayedWETH` proxy for the L2 chain that the
+# deploy config configures.
+#
+# The dispute game will be configured per the deploy config that is configured in `.env`, with some exceptions:
+# 1. The game type is set to `3`, or the `ASTERISC_KONA` game type.
+# 2. The absolute prestate is pulled from the kona release used for `op-contracts/v1.9.0-rc.3`
+#    (`kona-client-v0.1.0-beta.5`)
+./deploy.sh
+
+# 4. (Optionally) Generate a `superchain-ops` upgrade task.
+#
+# This script will consume the `deployments.json` file generated by `deploy.sh` and create an upgrade
+# bundle + superchain-ops documentation for the upgrade.
+./sc-ops.sh ./deployments.json
```

Comment:
Is this task effectively the same as would be generated by https://github.com/ethereum-optimism/superchain-ops/tree/main/tasks/sep/fp-recovery/005-set-game-implementation which is basically just a `setImplementation` call and reinit the AnchorStateRegistry?  If so, I wonder if we're better off just pointing to that task generator rather than duplicating the code? (I could be swayed either way though).

Reply:
Yeah it's similar - reinit ASR, set anchor state, then register the new game type. would pref to not muck around w/ pulling in superchain-ops as a dep for this branch though as there are some cyclical dependencies.

Reply:
No I just mean remove it from here and don't pull in superchain-ops or anything. You just do the deployment with this and the activation with superchain-ops template.  Typically deployment and activation should be separate steps anyway since you do reviews (and potentially gov approval) in between anyway.